sw/device/tests/aon_timer_wakeup_irq_test.c

```diff
@@ -0,0 +1,162 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/freestanding/limits.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/dif/dif_rv_timer.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+#include "sw/device/lib/testing/test_framework/test_status.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+const test_config_t kTestConfig;
+
+static const uint32_t kPlicTarget = kTopEarlgreyPlicTargetIbex0;
+static const uint32_t kTickFreqHz = 1000 * 1000;  // 1Mhz / 1us
+static const uint32_t kSleepTime_us = 10 * 1000;  // 10ms
+static dif_aon_timer_t aon_timer;
+static dif_rv_timer_t rv_timer;
```

Comment:
Why have you decided to use rv_timer instead of `ibex_mcycle_read`? If there is a good reason, that should be documented in detail - otherwise it is not immediately obvious.

"This version of the IP is fixed to have one Hart and one Timer per Hart."
If you must use it, you should make sure that you don't accidentally reconfigure the timer when it has already been configured. For example, by OTTF (however, I think that could potentially only be a problem for parallel tests - where OTTF uses tick to switch contexts).

Reply:
The reason is that the `mcycle` counter stops counting when the `wfi` is called, which makes sense because `mcycle` counts instructions cycles executed. The best approach then is to use the `mtime` register, which is basically attached to rv_timer in the opentitan.
https://docs.opentitan.org/hw/ip/rv_timer/doc/

Reply:
done!